www/benr01/js/calculator.js

```diff
@@ -402,13 +390,13 @@ function fillDropDownMenu(){
         var to_remove = document.getElementById("no_calculation_option");
         to_remove.remove(to_remove.selectedIndex);
     }
-    for (var i = element_calculator_select.length; i < localStorage.length; i++) { 
-        var option = document.createElement("option");
-        var stored_option = JSON.parse(localStorage.getItem(i));
 
-        option.text = stored_option.timestamp;
+    var stored_calculations = JSON.parse(localStorage.getItem("calculations"));
+    element_calculator_select.length = 0;
+    for (var i = 0; i < stored_calculations.length; i++) { 
+        var option = document.createElement("option");
+        option.text = stored_calculations[i].timestamp;
         option.value = i;
-
         element_calculator_select.add(option);
```

Comment:
tady jsem myslel to, ze tady opakovane pridavas ty optiony do toho selektu v cyklu, tzn v za prvni iteraci se tam prida jeden option, za druhou iteraci se tam prida druhy option, a po kazdem takovem tom kroku se musi prohlizec prekreslovat, reseni spociva v tom vsechny ty optiony pridat najednou, takze nejprve vygenerujes vsechny optiony v pameti, a pak pomoci jednoho prikazu je tam vsechny nacpes najednou.

priprava vice optionu neni mozne pomoci document.createElement, dela se to tak, ze ty optiony vygenerujes v HTML stringu, napr. 
```js
var options = '<option>1</option><option>2</option><option>3</option>';
selectElement.innerHTML = options;
```

Reply:
Aha, jasný. Opraveno